src/Components/Web.JS/src/Rendering/Renderer.ts

```diff
@@ -11,24 +11,38 @@ interface BrowserRendererRegistry {
 const browserRenderers: BrowserRendererRegistry = {};
 let shouldResetScrollAfterNextBatch = false;
 
-export function attachRootComponentToLogicalElement(browserRendererId: number, logicalElement: LogicalElement, componentId: number): void {
+export function attachRootComponentToLogicalElement(browserRendererId: number, logicalElement: LogicalElement, componentId: number, appendContent: boolean): void {
   let browserRenderer = browserRenderers[browserRendererId];
   if (!browserRenderer) {
     browserRenderer = browserRenderers[browserRendererId] = new BrowserRenderer(browserRendererId);
   }
 
-  browserRenderer.attachRootComponentToLogicalElement(componentId, logicalElement);
+  browserRenderer.attachRootComponentToLogicalElement(componentId, logicalElement, appendContent);
 }
 
 export function attachRootComponentToElement(elementSelector: string, componentId: number, browserRendererId?: number): void {
-  const element = document.querySelector(elementSelector);
+  const [selector, pseudoSelector] = elementSelector.split('::');
```

Comment:
I think we might need to be more restrictive in how we detect the pseudoselector. It would be a breaking change. 

Today we support selectors like “.myelems::first-of-type .another”.

Would it work to just detect “ends with ::after” but not otherwise look for :: appearing anywhere?

Sorry I was wrong saying we could throw if there are unrecognised pseudoselectors. We can’t if we want to continue allowing things like ::last-child etc. 

I think it will be ok to just recognise ::after at the very end of the string. It would also be nice to specifically recognise ending with ::before and throw since otherwise people might get confused. Those two didn’t have any supported meaning before so that’s non breaking. But other pseudoselectors could be ignored. 

Does that sound reasonable?

Reply:
Yep, sounds reasonable to me. I'll change this 😃